### PR TITLE
@eessex => Fade images in on load

### PR DESCRIPTION
**Pull request by damassi**
To get around the size-me flicker, this adds a slight fade to image load similar to how we currently do it. 

Addresses: 
- https://github.com/artsy/publishing/issues/188
- https://github.com/artsy/publishing/issues/157

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -11,6 +11,6 @@
     "dist/": true
   },
   "editor.tabSize": 2,
-  "editor.formatOnSave": true,
+  "editor.formatOnSave": false,
```
Comment (damassi) [inline .vscode/settings.json]: For those who like this, feel free to update VSCode editor settings via `cmd + ,` -> 'formatOnSave'.

File: src/Components/Publishing/Sections/ImageWrapper.tsx [hunk 2]
```diff
@@ -13,21 +14,77 @@ interface ImageWrapperProps extends React.HTMLProps<HTMLImageElement> {
...
-
-  return (
-    <StyledImageWrapper>
-      <BlockImage {...blockImageProps} />
-      {viewFullscreen}
-    </StyledImageWrapper>
-  )
+
+
+export class ImageWrapper extends React.Component<Props, any> {
```
Comment (damassi) [inline src/Components/Publishing/Sections/ImageWrapper.tsx]: All images that use `<ImageWrapper />` now fade in. We can update this at some later point

Comment (damassi) [reply src/Components/Publishing/Sections/ImageWrapper.tsx]: Also, i skipped the library route to keep this simple

File: src/Components/Publishing/Sections/ImageCollection.tsx [hunk 3]
```diff
@@ -20,83 +20,82 @@ interface ImageCollectionProps {
...
-      >
-        {renderedImage}
-      </ImageCollectionItem>
+      <ImageCollectionContainer>
+        {renderedImages}
+      </ImageCollectionContainer>
     )
-  })
-  return renderedImages
+  }
```
Comment (damassi) [inline src/Components/Publishing/Sections/ImageCollection.tsx]: DRYed this and the below up a bit so we don't have to pass stuff from function to function to function. It's less error prone and maintainable to just pluck from props.

File: src/Components/Publishing/Byline/Byline.tsx [hunk 4]
```diff
@@ -15,15 +15,16 @@ interface BylineContainerProps {
...
   const layout = props.layout || article.layout
   const title = article.social_title || article.thumbnail_title
   const url = articleFullHref(article.slug)
 
+  // TODO: Replace `authors` with `contributing_authors` when ready
   return (
     <BylineContainer layout={layout}>
       <Author
-        authors={authors}
+        authors={contributing_authors}
```
Comment (kanaabe) [inline src/Components/Publishing/Byline/Byline.tsx]: Thank you!

File: src/Components/Publishing/Sections/ImageWrapper.tsx [hunk 5]
```diff
@@ -47,6 +96,15 @@ const StyledImageWrapper = styled.div`
       opacity: 1;
     }
   `}
+
+  .BlockImage__container {
+    opacity: 0;
+    transition: opacity 1.0s;
+  }
```
Comment (kanaabe) [inline src/Components/Publishing/Sections/ImageWrapper.tsx]: Definitely not necessary but you could select with: ${BlockImage} {...}

File: src/Components/Publishing/Sections/ImageWrapper.tsx [hunk 6]
```diff
@@ -13,21 +14,69 @@ interface ImageWrapperProps extends React.HTMLProps<HTMLImageElement> {
...
+    try {
+      const imgTag = ReactDOM.findDOMNode(this.image)
+      const imgSrc = imgTag.getAttribute('src')
+      img.src = imgSrc
+    } catch (error) {
+      img.src = ''
+    } finally {
+      img.onload = this.onImageLoad
+      this.mounted = true
+    }
```
Comment (kanaabe) [inline src/Components/Publishing/Sections/ImageWrapper.tsx]: Cool!

**Comment by kanaabe** [comment]
Alrighty -- this looks good to go! Thanks for handling the last minute authors request!

**Pull request merged**